allennlp/data/dataset_readers/segmental_conll2000.py

```diff
@@ -0,0 +1,218 @@
+        outside of a segment. In the latter, U denotes a segment of length 1, and L is a tag end a
+        segment.
+    max_span_width: ``int``, optional (default=100)
+        Maximum width of chunk segments to consider. Any chunk longer than this is split up.
+    max_train_sentence_len: ``int``, optional (default=100)
+        Maximum length of training sentence to consider, any sentence longer than this is skipped.
+    label_namespace: ``str``, optional (default=``labels``)
+        Specifies the namespace for the chosen ``tag_label``.
+    use_segmental_labels: ``bool``, optional (default=`False`)
+        Labels for segmental layers, hence don't need BIO scheme.
+    use_binary_labels: ``bool``, optional (default=`False`)
+        0-1 labels to indicate segment boundaries.
+    """
+
+    def __init__(self,
+                 lazy: bool = True,
+                 token_indexers: Dict[str, TokenIndexer] = None,
+                 coding_scheme: str = "BIOUL",
+                 max_span_width: int = 100,
+                 max_train_sentence_len: int = 100,
+                 label_namespace: str = "labels",
+                 use_segmental_labels: bool = False,
+                 use_binary_labels: bool = False) -> None:
+        super().__init__(lazy)
+        self._token_indexers = token_indexers or {'tokens': SingleIdTokenIndexer()}
+
+        if coding_scheme not in ("BIO", "BIOUL"):
+            raise ConfigurationError("unknown coding_scheme: {}".format(coding_scheme))
+        self.coding_scheme = coding_scheme
+        self._original_coding_scheme = "BIOUL"
+
+        self._max_span_width = max(max_span_width, max_train_sentence_len)
+        self._max_train_sentence_len = max_train_sentence_len
+
+        self.label_namespace = label_namespace
+        self.use_segmental_labels = use_segmental_labels
+        self.use_binary_labels = use_binary_labels
+
+    @overrides
+    def _read(self, file_path: str) -> Iterable[Instance]:
+        # if `file_path` is a URL, redirect to the cache
+        file_path = cached_path(file_path)
+
+        with open(file_path, 'r') as data_file:
+            logger.info('Reading instances from lines in file at: %s', file_path)
+
+            # Group into alternative divider / sentence chunks.
+            for is_divider, lines in itertools.groupby(data_file, _is_divider):
+                # Ignore the divider chunks, so that `lines` corresponds to the words
+                # of a single sentence.
+                if not is_divider:
+                    fields = [line.strip().split() for line in lines]
+                    # unzipping trick returns tuples, but our Fields need lists
+                    tokens, _, chunk_tags = [list(field) for field in zip(*fields)]
+
+                    # HACK: Ignore long training sentences for memory efficiency.
+                    if "train" in file_path and len(tokens) > self._max_train_sentence_len:
```

Comment:
The way I'm doing this is just to override `_max_train_sentence_len` when I run evaluate. This more robust than this hack. Particularly because it's really obvious when you're modifying the default.

Reply:
I agree your suggestion is much better than this hack, but  I wrote this for the case where I was using a validation set during training, where that logic won't work?

Reply:
Oh, interesting! I actually wasn't using a validation set as the sampled softmax code defaulted to doing a real softmax when not training which caused OOMs. I'm curious, what are you using as your validation set?

Reply:
Sorry this was before I started working on the 1b benchmark, and was using PTB to build LMs.